system-contracts/scripts/preprocess-system-contracts.ts

```diff
@@ -17,6 +19,18 @@ async function preprocess(testMode: boolean) {
     params.SYSTEM_CONTRACTS_OFFSET = "0x9000";
   }
 
+  const timestampFilePath = path.join(process.cwd(), TIMESTAMP_FILE);
+  const folderToCheck = path.join(process.cwd(), CONTRACTS_DIR);
+
+  if ((await isFolderEmpty(OUTPUT_DIR)) || needsRecompilation(folderToCheck, timestampFilePath)) {
```

Comment:
not sure if it will work if:

1. I first compile in normal mode
2.I try to run the tests (for this the recompilation in `testMode` is done. 

Seems like on (2) the recompilation wont happen since no changes have been made to the contracts

Reply:
updated